blog/_src/posts/2016-11-16-constructive-galois-connections.md

```diff
@@ -0,0 +1,167 @@
+categories, functors and natural transformations.
+However, you can interpret the definition of adjunction in any
+"universe" that acts like the universe of categories, functors and
+natural transformations and it turns out we have such a universe.
+The universe I'm talking about is called \texttt{Rel}, consists of
+sets, relations between sets and *inclusion of relations*, i.e. that
+one relation is a subset of another.
+
+Then what does it mean to have an adjunction between two relations
+\\(R \subset X \times Y, Q \subset Y \times X\\)? Taking apart the
+definition it just means
+
+\\[ \Delta(X) \subset R;Q (1) \\]
+and
+\\[ Q;R \subset \Delta(Y) (2) \\]
+
+where \\(\Delta \\) means the *diagonal*, or equality relation on the set:
+
+\\[\Delta(X) = \{(x_1,x_2) \in X | x_1 = x_2 \} \\]
+
+So we just need to unravel what (1) and (2) mean above.  Unwinding
+(1), we get that for any \\(x \in X\\), there exists a \\(y \in Y \\)
+such that \\((x,y) \in R \\) and \\((y,x) \in Q\\). This tells us for
+one that \\(R \\) is a "right-total" relation and \\(Q \\) is a
+"left-total". Every \\(x \\) is related to some \\( y\\) by \\( R \\)
+and \\( Q\\).
+
+If we unwind (2), we get that for any \\(y,y' \in Y\\) if there's some
+\\(x \in X \\) such that \\((x,y) \in R \\) and \\((y',x) \in Q \\)
+then actually \\(y = y')\\). This one is a bit more mysterious, but
+first, let's see what this tells us about the relationship between
+\\(R\\) and \\(Q \\).
+
+If \\((x,y) \in R \\), then by (1) there's some \\(y' \in Y\\) so that
+\\((x,y') \in R \\) and \\((y',x) \in Q\\). Then, by (2) we know that
+\\(y = y'\\), so we've shown that if \\((x,y) \in R \\) then \\((y,x)
+\in Q\\). Then a completely symmetric argument shows that if \\((y,x)
+\in Q \\) then \\((x,y)\in R\\)! So we've discovered that actually
+\\(Q \\) is just the opposite relation of \\(R \\).
+
+Then if we look at (2) again but replace the \\(Q\\)'s by flipped
+\\(R\\)'s we get that for any \\(y,y' \in Y\\), if there's some \\(x
+\in X\\) such that \\((x,y) \in R \\) and \\((x,y')\in R\\) then \\(y
+= y'\\), which tells us that \\(R \\) is a partial function, i.e.,
+that every \\(x \\) is related to at most one \\(y \\) by \\(R \\).
+
+You may recognize it now, our \\(R \subset X \times Y \\) is just a
+function, and saying \\(R, Q\\) are adjoint is exactly the same as
+saying that \\(Q = R^{\text{op}}\\) and \\(R \\) is a function.
+Adjunctions are so pervasive you saw them when you were still wearing
+mathematical diapers.
+
+# Constructive Galois Connections
+
+Back to constructive Galois connections, I hope if you read the paper
+you can see that their theorem is a generalization of the above
+argument, where instead of relations "monotone relations", i.e.,
+downward-closed \\(R \subset A^{\text{op}} \times B \\). Then you can
+interpret the definition of adjunction in that universe and get that
+it's the same as a Kleisli Galois connection and that a similar
```

Comment:
They are the same as kleisli galois connections except those are in terms of functions to a powerset instead of relations, and then the proof shows that kleisli gcs are the same as constructive gcs.